code/modules/lobby_cam/lobby_eye_ss.dm

```diff
@@ -0,0 +1,117 @@
+	///the pathway that the camera will take
+	var/list/pathway = list()
+
+/datum/controller/subsystem/lobby_eye/Initialize(start_timeofday)
+	. = SS_INIT_SUCCESS
+	linked_camera = new()
+	fading_screen = new()
+	fire()
+
+//fades out the artea logo in the bottom left
+/datum/controller/subsystem/lobby_eye/proc/fade_logo()
+	for(var/mob/checking_mob as anything in GLOB.new_player_list)
+		var/atom/movable/screen/artea_logo/logo_screen = locate() in checking_mob.client.screen
+		if(logo_screen?.alpha == 255)
+			animate(logo_screen, alpha = 0, time = 5 SECONDS)
+
+//gets everyone on the camera
+/datum/controller/subsystem/lobby_eye/proc/lock_eyes()
+	for(var/mob/checking_mob as anything in GLOB.new_player_list)
+		if(!checking_mob.client)
+			continue
+		if(!checking_mob.client?.prefs.read_preference(/datum/preference/toggle/lobby_cam))
+			continue
+		if(SSticker.current_state < GAME_STATE_PREGAME)
+			continue
+		if(!(fading_screen in checking_mob.client?.screen))
+			checking_mob.client?.screen += fading_screen
+		if(checking_mob.client?.eye != linked_camera)
+			checking_mob.client?.eye = linked_camera
+		if(checking_mob.lighting_alpha != LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE)
+			checking_mob.lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+
+//gets everyone off of the camera
+/datum/controller/subsystem/lobby_eye/proc/unlock_eyes()
+	for(var/mob/checking_mob in GLOB.new_player_list)
+		if(!checking_mob.client)
+			continue
+		if(fading_screen && (fading_screen in checking_mob.client?.screen))
+			checking_mob.client?.screen -= fading_screen
+		if(checking_mob.client?.eye != linked_camera)
+			continue
+		checking_mob.client?.eye = checking_mob.client?.mob
+
+/datum/controller/subsystem/lobby_eye/fire(resumed)
+	//the config
+	if(!CONFIG_GET(flag/lobby_camera))
+		return
+
+	//fade out the logo and unlock eyes when 10 seconds until round start
+	var/time_remaining = SSticker.GetTimeLeft()
+	if(time_remaining <= 20 SECONDS && SSticker.current_state == GAME_STATE_PREGAME)
+		unlock_eyes()
+		fade_logo()
+		return
+
+	//we should only work in pregame
+	if(SSticker.current_state != GAME_STATE_PREGAME)
+		unlock_eyes()
+		return
+
+	//find the starting location
+	var/turf/starting_turf = get_safe_random_station_turf()
+	pathway = list(starting_turf)
+
+	//start building the pathway
+	var/chosen_direction = pick(GLOB.cardinals)
+	var/invalid_crossed = 0
+	var/turf/chosen_turf = starting_turf
+	for(var/creating_pathway in 1 to path_target)
+		chosen_turf = get_step(chosen_turf, chosen_direction)
+		if(!chosen_turf)
+			break
+		if(is_type_in_list(get_area(chosen_turf), invalid_areas))
+			invalid_crossed++
+		if(invalid_crossed >= invalid_reset)
+			break
+		pathway += chosen_turf
+
+	//move camera to starting location and lock
+	linked_camera.loc = starting_turf
+	lock_eyes()
+
+	//fade in
+	for(var/fade_in = 1, fade_in < 11, fade_in++)
+		fading_screen.icon_state = "[fade_in]"
+		sleep(1)
+	fading_screen.icon_state = "11"
```

Comment:
why not use a an animate easing

Reply:
yea literally replace all this easing in icon state stuff with animate() calls to save server cpu and network cause this looks e x p e n s i v e to do serverside when it could just be done on the client and require only two icon states